PocketKit/Sources/Sync/Operations/FetchList.swift

```diff
@@ -73,6 +75,31 @@ class FetchList: SyncOperation {
         initialDownloadState.send(.completed)
     }
 
+    private func fetchTags() async throws {
+        guard lastRefresh.lastRefresh == nil else { return }
+
+        var shouldFetchNextPage = true
+        var pagination = PaginationInput()
+
+        while shouldFetchNextPage {
+            let query = TagsQuery(pagination: pagination)
+            let result = try await apollo.fetch(query: query)
+
+            result.data?.user?.tags?.edges?.forEach { edge in
+                guard let node = edge?.node else { return }
+                let tag = space.fetchOrCreateTag(byName: node.name)
+                tag.update(remote: node.fragments.tagParts)
+            }
+
+            if let pageInfo = result.data?.user?.tags?.pageInfo {
+                pagination.after = pageInfo.endCursor
+                shouldFetchNextPage = pageInfo.hasNextPage
+            } else {
+                shouldFetchNextPage = false
+            }
+        }
+    }
+
```

Comment:
This is the main thrust of this change

Reply:
Appreciate the guidance: I was drowning in the generated code. 🏊‍♀️

Reply:
Will this cover the edge case where …

user’s iOS device goes offline;
on web, user saves item;
on web, user tags item with new tag “kayak”;
on web, user deletes item;
user’s iOS device goes back online

Will iOS still download “kayak” tag?

Reply:
Oh shoot. No it wouldn't.

Reply:
Okay, let's move forward with this, and then discuss that edge case separately.